pkg/client/cli/intercept/state.go

```diff
@@ -149,7 +150,10 @@ func create(sif State, ctx context.Context) (acquired bool, err error) {
 		// local-only
 		return true, nil
 	}
-	fmt.Fprintf(s.cmd.OutOrStdout(), "Using %s %s\n", r.WorkloadKind, s.AgentName)
+	detailedOutput := s.DetailedOutput && output.WantsFormatted(s.cmd)
+	if !detailedOutput {
+		fmt.Fprintf(s.cmd.OutOrStdout(), "Using %s %s\n", r.WorkloadKind, s.AgentName)
+	}
```

Comment:
Instead of requiring that both flags are set, if `DetailedOutput` can we default to yaml?